### PR TITLE
[Chore] Add ECS iam management

### PR DESCRIPTION
**Pull request by samkahchiin**
### Summary
Module for ECS IAM policy.

### CODE REVIEW

File: iam/iam-policy-for-ecs/main.tf [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+    local.staging_read_access_projects,
+    local.production_read_access_projects
+  )
+
+  write_access_projects = concat(
+    local.staging_write_access_projects,
+    local.production_write_access_projects
+  )
+
+  region = "eu-central-1"
```
Comment (samkahchiin) [inline iam/iam-policy-for-ecs/main.tf:35]: Hardcoded this value for now but will think about how to manage this dynamically.
Perhaps we can take the assignment of policy to the user from the module.

Comment (swiknaba) [reply iam/iam-policy-for-ecs/main.tf:35]: @marcqualie should we prepend the region to the policy names, that we need for each geo region?

File: iam/iam-policy-for-ecs/taggable-resources/variables.tf [hunk 2]
```diff
@@ -0,0 +1,17 @@
+variable "region" {
```
Comment (swiknaba) [inline iam/iam-policy-for-ecs/taggable-resources/variables.tf:1]: can we add a README to this? :-)

File: iam/iam-policy-for-ecs/core/main.tf [hunk 3]
```diff
@@ -0,0 +1,103 @@
+data "aws_iam_user" "main" {
+  user_name = var.username
+}
+
+locals {
+  staging_developer_access_projects    = try(data.aws_iam_user.main.tags.staging-developer-access-projects, "")
```
Comment (swiknaba) [inline iam/iam-policy-for-ecs/core/main.tf:6]: is it possible to keep this general, i.e. not bound to environments named `staging` and `production`?

Comment (samkahchiin) [reply iam/iam-policy-for-ecs/core/main.tf:6]: Can do if we pass the projects and environment as variables.
The tradeoff is we can't automatically assign the permission based on the user tags using the modules
(We have to extract the values and passing them as variables into the modules)
Both have its pros and cons
Option 1: Extracting the tag values inside module -> Simplified the usage of module, i.e. just need to pass the username into the module, less flexibility
 Option 2: Making the variable more general -> More flexibility, need to extract the values of projects name and environment from tag (if relevant) and pass into the modules

**Review by swiknaba — APPROVED** [review]
Tested on our DBL Test project. works 👍 

Thanks 🎉

**Pull request merged**